16.0[IMP] account_loan: Currency on loan not readonly + convert credit/debit for different currencies

lmarion-source opened the pull request:
Business case: 

- Need to change the currency on a loan if different from the one on the journal/company
- If the journal on the loan is - for instance - in EUR, the loan also in EUR, but the company in USD, when we create the moves the debit/credit on lines stays in EUR. 
When we create move lines from a journal entry, if the journal is in EUR but the company in USD, the credit/debit is in USD and amount_currency is filled in EUR. 
For the loans, we wanted the same behavior : when creating the move lines from the loan line, the credit/debit should be in the currency of the company while the  amount_currency is filled with the value in the journal currency

lmarion-source committed:
[IMP] account_loan: Allow to modify currency_id on account loan

etobella commented on account_loan/models/account_loan.py:
```diff
@@ -163,7 +163,7 @@ def _default_company(self):
         "res.currency",
         compute="_compute_currency",
-        readonly=True,
+        readonly=False,
```
You probably need to make it stored, isn't it?

acsonefho commented on account_loan/tests/test_loan.py:
```diff
@@ -617,3 +617,33 @@ def create_loan(self, type_loan, amount, rate, periods):
+        for line in loan.line_ids:
+            self.assertEqual(line.currency_id, loan.currency_id)
+
+        line = loan.line_ids.filtered(lambda r: r.sequence == 1)
```
Any constraint say that the `sequence` is unique. If you want the first line: `fields.first(loan.line_ids)`

acsonefho commented on account_loan/views/account_loan_view.xml:
```diff
@@ -149,7 +149,7 @@
                                     />
                                     <field name="interest_expenses_account_id" />
-                                    <field name="currency_id" invisible="1" />
+                                    <field name="currency_id" />
```
`<field name="currency_id" groups="base.group_multi_currency" />`

acsonefho commented on account_loan/tests/test_loan.py:
```diff
@@ -617,3 +617,33 @@ def create_loan(self, type_loan, amount, rate, periods):
+
+        move_lines = line.mapped("move_ids.line_ids")
+        # Credit/Debit lines should be in USD, while amount_currency is in EUR
+        self.assertEqual(
```
`self.assertAlmostEqual(amount1, amount2, places=xxx)`

etobella reviewed (commented):
Technically looks OK. Please use proper commit name and leave just one commit please. After that we can proceed with merge.

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message

etobella commented on account_loan/models/account_loan_line.py:
```diff
@@ -317,15 +342,31 @@ def _add_long_term_account_values(self, vals):
             vals.append(
                 {
-                    "account_id": self.long_term_loan_account_id.id,
+                    "account_id": self.loan_id.short_term_loan_account_id.id,
```
This shouldn't be changed!

etobella reviewed (changes requested):
Technically looks OK. Please use proper commit name and leave just one commit please. After that we can proceed with merge.

https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message

lmarion-source commented:
> Technically looks OK. Please use proper commit name and leave just one commit please. After that we can proceed with merge.
> 
> https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#commit-message


I changed the commit message, however, I let 2 commits because it's 2 different things : currency not readonly, then convert credit/debit for different currencies.

etobella commented:
The commit message must include the module name.

lmarion-source committed:
[IMP] account_loan: Currency not readonly + convert in company currency

    1. We allow to change the currency on the loan if the user needs it

    2. If the currency on the loan (or on the journal) is different from the one on the company, we do the conversion for credit/debit + add amount in currency on the move line

petrus-v reviewed (approved):
Looks good to me

AnizR reviewed (approved):
LGTM 👍

etobella reviewed (approved):
LGTM

/ocabot merge patch
